### PR TITLE
Deactivate tests using deprecated numpy syntax

### LINKED ISSUE #1272: Tests don't run with numpy 1.24.0
### Describe the bug
Numpy has been warning for a while that it will deprecate `np.bool`, `np.int`, `np.float`, etc. It seems that as of 1.24.0 this is now deprecated. As a result some of the tests now fail.

### To Reproduce
E.g. : https://github.com/pyccel/pyccel/blob/master/tests/epyccel/recognised_functions/test_numpy_types.py#L145

### Expected behavior
We should remove deprecated behaviour or prevent these tests from running if a more recent numpy version is used

### PR DESCRIPTION
**Pull request by EmilyBourne**
- Deactivate tests that use deprecated numpy syntax, fixes #1272 
- Correct tests to use bound that don't generate `RuntimeWarning` messages about overflows
- Make type printing match python completely
- Stop deleting `.lock` file during folder overwrite (potential cause of mac parallel bug)

### CODE REVIEW

File: tests/epyccel/recognised_functions/test_numpy_types.py [hunk 1]
```diff
@@ -292,15 +295,21 @@ def get_int8(a):
...
 
     integer8 = randint(min_int8, max_int8, dtype=np.int8)
-    integer16 = randint(min_int16, max_int16, dtype=np.int16)
-    integer = randint(min_int, max_int, dtype=int)
-    integer32 = randint(min_int32, max_int32, dtype=np.int32)
-    integer64 = randint(min_int64, max_int64, dtype=np.int64)
+    integer16 = randint(min_int8, max_int8, dtype=np.int16)
+    integer = randint(min_int8, max_int8, dtype=int)
+    integer32 = randint(min_int8, max_int8, dtype=np.int32)
+    integer64 = randint(min_int8, max_int8, dtype=np.int64)
```
Comment (bauom) [inline tests/epyccel/recognised_functions/test_numpy_types.py:312]: shouldn't these use min and max for the associated types?

File: tests/epyccel/recognised_functions/test_numpy_types.py [hunk 2]
```diff
@@ -652,10 +666,10 @@ def test_numpy_float_scalar(language, get_float):
     integer32 = randint(min_int32, max_int32, dtype=np.int32)
     integer64 = randint(min_int64, max_int64, dtype=np.int64)
 
-    fl = uniform(min_float / 2, max_float / 2)
+    fl = uniform(min_float32 / 2, max_float32 / 2)
     fl32 = uniform(min_float32 / 2, max_float32 / 2)
     fl32 = np.float32(fl32)
-    fl64 = uniform(min_float64 / 2, max_float64 / 2)
+    fl64 = uniform(min_float32 / 2, max_float32 / 2)
```
Comment (bauom) [inline tests/epyccel/recognised_functions/test_numpy_types.py:672]: why are we using `float32` instead of `float64` in these examples?

**Review by bauom — CHANGES_REQUESTED** [review]

File: tests/epyccel/recognised_functions/test_numpy_types.py (hunk 1 above)
Comment (EmilyBourne) [reply tests/epyccel/recognised_functions/test_numpy_types.py:312]: No. The test converts integers of precision X to int8 objects. If the original int is larger than the target int then there will be overflow warnings. The latest version of numpy flags these overflow warnings with a `DeprecationError`, indicating that in a few versions time, this code will fail if we pass an int that is out of the bounds `[min_int8, max_int8]` to the function `int8`

File: tests/epyccel/recognised_functions/test_numpy_types.py (hunk 2 above)
Comment (EmilyBourne) [reply tests/epyccel/recognised_functions/test_numpy_types.py:672]: As above. We are converting to  `float32`, so the `float64` can't contain a float which is too big to be cast without overflow/underflow errors

File: tests/epyccel/recognised_functions/test_numpy_types.py (hunk 1 above)
Comment (bauom) [reply tests/epyccel/recognised_functions/test_numpy_types.py:312]: ah ok i see :+1:

**Review by bauom — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

**Pull request merged**